lib/ui/views/homepage/homepage_viewmodel.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:receipe_app/app/app.locator.dart';
+import 'package:receipe_app/data_model/product_model.dart';
+import 'package:receipe_app/ui/common/app_images.dart';
+import 'package:stacked/stacked.dart';
+import 'package:stacked_services/stacked_services.dart';
+
+class HomepageViewModel extends BaseViewModel {
+  final _navigationService = locator<NavigationService>();
+
+  List<ProductModel> productItems() {
+    return [
+      ProductModel(
+        date: 'May 28, 2023',
+        image: AppImages.spagLog,
+        title: 'Spaghetti',
+        isLiked: false,
+      ),
+      ProductModel(
+        isLiked: false,
+        date: 'May 28, 2023',
+        image: AppImages.spagLogTwo,
+        title: 'Spaghetti',
+      ),
+      ProductModel(
+        isLiked: false,
+        date: 'May 28, 2023',
+        image: AppImages.spagLogThree,
+        title: 'Spaghetti',
+      ),
+      ProductModel(
+        isLiked: true,
+        date: 'May 28, 2023',
+        image: AppImages.spagLogFour,
+        title: 'Spaghetti',
+      ),
+      ProductModel(
+        isLiked: true,
+        date: 'May 28, 2023',
+        image: AppImages.spagLog,
+        title: 'Spaghetti',
+      ),
+      ProductModel(
+        isLiked: false,
+        date: 'May 28, 2023',
+        image: AppImages.spagLogTwo,
+        title: 'Spaghetti',
+      ),
+    ];
+  }
```

Comment:
Simply store in a list variable, using a function here isn't ideal. eg
```
List<ProductModel> productList = [
  ProductModel();
];
```

Reply:
Is there a particular reason thou?
this method also iterate without an issue.

Reply:
an analogy would be taking medications for the side effect and not the proper usecase. Variables are for stores data, List variables are for storing list, so why use a function to return a list when you can store variables in a list ??